worker/worker-factory.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const DB_WORKER_ACTIONS = require(
+  './db-worker-actions/db-worker-actions.const'
+)
+
+module.exports = (executeAction) => {
+  const { workerData, parentPort } = require('worker_threads')
+  const Database = require('better-sqlite3')
+
+  const _connect = (args) => {
+    const {
+      dbPath = './sqlite-db.db',
+      readonly = false,
+      fileMustExist = false,
+      timeout = 5000,
+      verbose = false
+    } = args
+
+    return new Database(
+      dbPath,
+      {
+        readonly,
+        fileMustExist,
+        timeout,
+        ...(verbose ? { verbose: console.log } : {})
+      }
+    )
+  }
+
+  const _serializeError = (err) => {
+    if (!(err instanceof Error)) {
+      return err
+    }
+
+    return Object.keys(err).reduce((obj, key) => {
+      obj[key] = err[key]
+
+      return obj
+    }, {
+      name: err.name,
+      message: err.message,
+      stack: err.stack,
+      isError: true
+    })
+  }
+
+  const db = _connect(workerData)
+
+  parentPort.on('message', (args) => {
```

Comment:
did you do some benchmarking, for lets say 500k writes and 500k reads and compared it with the our other fac? switching the context and keeping the workers also has some overhead and we want to make sure that we improve perf

Reply:
Currently, we are in progress in reports tool, doing a migration to this driver and also will leave previously db implementation to have possible to benchmark
The final result we can see on reports tool with adding some test
Some benchmark is already done in better-sqlite3 (but without worker threads) https://github.com/JoshuaWise/better-sqlite3/blob/master/docs/benchmark.md